chalice/logs.py

```diff
@@ -107,3 +127,105 @@ def retrieve_logs(self, include_lambda_messages=True, max_entries=None):
             shown += 1
             if max_entries is not None and shown >= max_entries:
                 return
+
+
+class BaseLogEventGenerator(object):
+    def __init__(self, client):
+        # type: (TypedAWSClient) -> None
+        self._client = client
+
+    def iter_log_events(self, log_group_name, options):
+        # type: (str, LogRetrieveOptions) -> Iterator[CWLogEvent]
+        pass
+
+
+class LogEventGenerator(BaseLogEventGenerator):
+    def iter_log_events(self, log_group_name, options):
+        # type: (str, LogRetrieveOptions) -> Iterator[CWLogEvent]
+        logs = self._client.iter_log_events(log_group_name=log_group_name,
+                                            start_time=options.start_time)
+        for log_message in logs:
+            yield log_message
+
+
+class FollowLogEventGenerator(BaseLogEventGenerator):
+
+    _POLL_TIME = 5
+
+    def __init__(self, client, sleep=time.sleep, poll_time=_POLL_TIME):
+        # type: (TypedAWSClient, Callable[[int], None], int) -> None
+        self._client = client
+        self._sleep = sleep
+        self._event_id_cache = defaultdict(
+            set)  # type: Dict[datetime, Set[str]]
+        self._poll_time = poll_time
+
+    def iter_log_events(self, log_group_name, options):
+        # type: (str, LogRetrieveOptions) -> Iterator[CWLogEvent]
```

Comment:
I wonder if this makes sense to hoist this specialized logic to the options class? The option takes in `follow` as a parameter so we should be able to set 10 minutes in the past there if it is `--follow` and no `--since` was specified.

Reply:
Yeah I can do that.  Personally I'd like to just update the default to be 10 minutes in the past for both cases.  Once your app has been deployed and running for a while showing all the logs for `chalice logs` is almost not what you want.  I worry it's going to be confusing to have different defaults between `--follow` vs. `--no-follow`.

Reply:
>> Personally I'd like to just update the default to be 10 minutes in the past for both cases.

I think I would lean to that opinion too. I always thought it was a bit jarring that the `logs` command showed all of the logs, as made it harder to figure out if any new logs were produced.

Once this flag is merged, I'd suspect most people will use the `--follow` flag for active debugging. Then, I suspect the `logs` command without `--follow` will be more for postmortems where you will likely be wanting all logs or all logs from a particular date so I can see the default of all logs making sense there. So I'm not sure if will be too big of a deal. I think if the command was named `tail` instead of `logs` it would definitely be a bit more confusing, especially if you compare it to a  Unix `tail` command.